app/src/main/java/com/example/voice_recorder_app/AudioListFragment.kt

```diff
@@ -14,8 +12,10 @@ import androidx.recyclerview.widget.RecyclerView
 import com.google.android.material.bottomsheet.BottomSheetBehavior
 import java.io.File
 
-class AudioListFragment : Fragment() {
-
+class AudioListFragment : Fragment()  {
+    val path: String? = activity?.getExternalFilesDir("/")?.absolutePath
+    val directory = File(path).parentFile
+    val allFiles = directory.listFiles()

```

Comment:
```suggestion
    lateinit var path: String
    lateinit var directory
    lateinit var allFiles

```

Reply:
Then set these in the `onCreateView` like you had before, except don't declare the `val` part again:

```
path = activity?.getExternalFilesDir("/")?.absolutePath
... etc
```

Reply:
It worked after this modification! Thank you!!!!